snuba/cli/querylog_to_csv.py

```diff
@@ -0,0 +1,183 @@
+    WHERE (query_kind = 'Select')
+    AND (type = '{type}')
+    AND (databases = {databases})
+    AND (tables = {tables})
+    AND (query_start_time >= {start})
+    AND (query_start_time <= {end})
+    """
+
+
+def get_credentials() -> Tuple[str, str]:
+    # TOOO don't hardcode credentials, use settings
+    return ("default", "")
+
+
+@click.command()
+@click.option(
+    "--clickhouse-host",
+    help="Clickhouse server to write to.",
+    required=True,
+    default="localhost",
+)
+@click.option(
+    "--clickhouse-port",
+    type=int,
+    help="Clickhouse native port to write to.",
+    required=True,
+    default=9000,
+)
+@click.option(
+    "--database",
+    help="ClickHouse Database to query from.",
+    required=True,
+    default="default",
+)
+@click.option(
+    "--table",
+    help="ClickHouse tables to query from.",
+    required=True,
+)
+@click.option(
+    "--event-type",
+    help="Type of event that occured while executing query.",
+    type=click.Choice(
+        ["QueryFinish", "ExceptionBeforeStart", "ExceptionWhileProcessing"]
+    ),
+    required=True,
+    default="QueryFinish",
+)
+@click.option(
+    "--start-time",
+    help="Starting timestamp for the query",
+    required=True,
+)
+@click.option(
+    "--end-time",
+    help="Ending timestamp of the query",
+)
+@click.option("--log-level", help="Logging level to use.")
+def querylog_to_csv(
+    *,
+    clickhouse_host: str,
+    clickhouse_port: int,
+    table: str,
+    database: str,
+    event_type: str,
+    start_time: str,
+    end_time: Optional[str] = None,
+    log_level: Optional[str] = None,
+) -> None:
+    """
+    Use this command when you want to capture the results from the
+    clickhouse querylog after you have re-run queries (using the
+    `snuba query-replayer` command).
+    """
+    setup_logging(log_level)
+    setup_sentry()
+
+    # clickhouse expects tables like -> ['default.errors_local']
+    tables = [f"{database}.{table}"]
+
+    query = get_query_results(event_type, [database], tables, start_time, end_time)
+
+    (clickhouse_user, clickhouse_password) = get_credentials()
+    connection = ClickhousePool(
+        host=clickhouse_host,
+        port=clickhouse_port,
+        user=clickhouse_user,
+        password=clickhouse_password,
+        database=database,
+        client_settings=ClickhouseClientSettings.QUERY.value.settings,
+    )
+    results = connection.execute(query)
+    filename = format_filename(table)
+
+    # TODO: save this file to GCS bucket
+    write_querylog_results_to_csv(results.results, f"/tmp/{filename}.csv")
+
+    # TODO: maybe use new specific channel id
+    slack_client = SlackClient(
+        channel_id=settings.SNUBA_SLACK_CHANNEL_ID, token=settings.SLACK_API_TOKEN
+    )
+
+    slack_client.post_file(
+        file_name=f"{filename}.csv",
+        file_path=f"/tmp/{filename}.csv",
+        file_type="csv",
+        initial_comment=f"Querylog Result Report: {filename}",
+    )
```

Comment:
Let's make this optional as well. We only need slack notifications until we are confident of this tooling.

Reply:
not sure we even really need this slack output, just wanted to have something posted when testing this out